isthmus/src/main/java/io/substrait/isthmus/expression/LiteralConverter.java

```diff
@@ -155,7 +154,7 @@ public Expression.Literal convert(RexLiteral literal) {
         yield timestamp(n, ldt);
       }
       case INTERVAL_YEAR, INTERVAL_YEAR_MONTH, INTERVAL_MONTH -> {
-        var intervalLength = literal.getValueAs(BigDecimal.class).longValue();
+        long intervalLength = Objects.requireNonNull(literal.getValueAs(Long.class));
```

Comment:
I was curious why this change is okay but getValueAs() returns the object casted if the type matches, otherwise it has some conversion logic for some types, so here it'll first cast the object to BigDecimal and then call `.longValue()` on it, so basically what the previous code was doing 👍

Reply:
Actually the `literal.getValueAs(BigDecimal.class)` that was there previously calls `liternal.getValueAs(Long.class)` internally before then converting it into a BigDecimal... which we then convert back to a long. It seems more sensible to just ask for the type we want. Functionally it is the same.

Reply:
Hm, I think the getValueAs(Decimal) on a interval type would end up in this code at the beginning of `getValueAs`:
```
    if (value == null || clazz.isInstance(value)) {
      return clazz.cast(value);
    }
```
while now we're going to 
```
    case INTERVAL_YEAR:
...
} else if (clazz == Long.class) {
        return clazz.cast(((BigDecimal) value).longValue());
```

Reply:
Do you need me to change this back to we explicitly ask for a type we don't want, or is it OK to leave this change so we ask for the type we actually want and Calcite provides it in whichever way is best fits its current implementation?

Reply:
Oh sorry this is totally fine by me 👍